modal/_utils/async_utils.py

```diff
@@ -451,41 +458,124 @@ class NestedAsyncCalls(Exception):
     pass
 
 
-def run_generator_sync(
-    gen: typing.AsyncGenerator[YIELD_TYPE, SEND_TYPE],
-) -> typing.Generator[YIELD_TYPE, SEND_TYPE, None]:
-    try:
-        asyncio.get_running_loop()
-    except RuntimeError:
-        pass  # no event loop - this is what we expect!
-    else:
-        raise NestedAsyncCalls()
-    loop = asyncio.new_event_loop()  # set up new event loop for the map so we can use async logic
+class Runner:
```

Comment:
A "lightweight" `asyncio.Runner` (from python 3.11). Very similar to our `UserCodeEventLoop` class in `_container_entrypoint.py` which has a similar purpose.

Reply:
Just put the same (or at least very similar) backport into synchronicity (version tbd, 0.9?) so might wait with merging this until I can use that